boolean/boolean.go

```diff
@@ -14,3 +14,159 @@ func IsTrue(v string) bool {
 
 	return b
 }
+
+/*
+Toggle negates the given boolean value.
+
+Returns:
+  - true if the input is false.
+  - false if the input is true.
+*/
+func Toggle(value bool) bool {
+	return !value
+}
+
+/*
+AllTrue checks if all values in the slice are true.
+
+Returns:
+  - false if the slice is empty or contains at least one false value.
+  - true if all values are true.
+*/
+func AllTrue(values []bool) bool {
+	if len(values) == 0 {
+		return false // Empty slices are considered not "all true".
+	}
+
+	for _, b := range values {
+		if !b {
+			return false
+		}
+	}
+	return true
+}
+
+/*
+AnyTrue checks if at least one value in the slice is true.
+
+Returns:
+  - false if the slice is empty or contains no true values.
+  - true if at least one value is true.
+*/
+func AnyTrue(values []bool) bool {
+	if len(values) == 0 {
+		return false // Empty slices are considered not "any true".
+	}
+
+	for _, b := range values {
+		if b {
+			return true
+		}
+	}
+	return false
+}
+
+/*
+NoneTrue checks if none of the values in the slice are true.
+
+Returns:
+  - true if the slice is empty or contains no true values.
+  - false if at least one value is true.
+*/
+func NoneTrue(values []bool) bool {
+	if len(values) == 0 {
+		return true // Empty slices are considered "none true".
+	}
+
+	for _, b := range values {
+		if b {
+			return false
+		}
+	}
+	return true
+}
+
+/*
+CountTrue counts the number of true values in the slice.
+
+Returns:
+  - The number of elements in the slice that are true.
+*/
+func CountTrue(values []bool) int {
+	count := 0
+	for _, b := range values {
+		if b == true {
+			count++ // Increment for each true value.
+		}
+	}
+	return count
+}
+
+/*
+CountFalse counts the number of false values in the slice.
+
+Returns:
+  - The number of elements in the slice that are false.
+*/
+func CountFalse(values []bool) int {
+	count := 0
+	for _, b := range values {
+		if b == false {
```

Comment:
just `if !b` is fine too

Reply:
Can be covered as part of tech debt ticket.